x/mongo/driver/compression_test.go

```diff
@@ -40,3 +43,78 @@ func TestCompression(t *testing.T) {
 		})
 	}
 }
+
+func TestZstdWindowSize(t *testing.T) {
+	tests := []struct {
+		inputSize  int
+		level      zstd.EncoderLevel
+		windowSize int
+	}{
+		{
+			inputSize:  512,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 1024,
+		},
+		{
+			inputSize:  512000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 524288,
+		},
+		{
+			inputSize:  16000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+		{
+			inputSize:  32000000,
+			level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
+			windowSize: 16777216,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(strconv.Itoa(test.inputSize), func(t *testing.T) {
+			windowSize := calcZstdWindowSize(test.inputSize, test.level)
+			assert.Equal(t, test.windowSize, windowSize)
+		})
+	}
+}
+
+func BenchmarkCompression(b *testing.B) {
+	payload := func() []byte {
+		f, err := os.Open("compression.go")
```

Comment:
Are you compressing `compression.go` as the benchmark? Cool 😆

Reply:
I was trying to compress random bytes, but the compressor did poorly on them with compression ratio > 1. Also, compression.go is better than [Lorem ipsum](https://en.wikipedia.org/wiki/Lorem_ipsum), which I have to save the text somewhere else.